### PR TITLE
Front End For Settings and Entries

### PR DESCRIPTION
**Pull request by Benek6h**
# Description

This is a UX redesign of the Settings and Entries pages. Including dark and light mode.

# Fixes # (issue)
#108 

Please delete options that are not relevant.

- [x] UX Update

# How Has This Been Tested?

No tests for UX updates
### Another Update for Tests
When setting mock values for the settings in your test, make sure you set them before running app.main().
If you do not, then the current page won't redraw itself and will potentially cause problems

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Media
Light & Dark Mode Settings:
image
image

Light & Dark Mode Tags:
image
image

Light & Dark Mode Entries Panel:
Screenshot_2023-11-20-13-35-25-64_2354f2ed585dce91573efc8816656cc0
Screenshot_2023-11-20-13-42-47-36_2354f2ed585dce91573efc8816656cc0

### CODE REVIEW

**Comment by Benek6h** [comment]
The new entry button will be moved to the navigation bar, thus I did not edit it in this pr to fit the style conditions.

File: app/integration_test/entries_test.dart
Comment (S1robe) [inline app/integration_test/entries_test.dart:1]: Im assuming duplicate?

**Review by S1robe — APPROVED** [review]
Great additions, youve done a great job making the UI better.
Even added your own assets.

Thank you for helping out.

File: app/lib/uiwidgets/cards.dart [hunk 1]
```diff
@@ -45,100 +46,117 @@ class _DisplayCardState extends State<DisplayCard> {
...
+          ),
+
+          // Rounded background
           child: Container(
-            //width: MediaQuery.of(context).size.width,
+            padding: const EdgeInsets.all(10),
             decoration: BoxDecoration(
+              color: getCurrentTheme().colorScheme.background.withAlpha(200),
+              borderRadius: BorderRadius.circular(15),
               gradient: LinearGradient(
```
Comment (Cooljawty) [inline app/lib/uiwidgets/cards.dart:64]: I like this gradient much more. Great job!

File: app/lib/uiwidgets/cards.dart [hunk 2]
```diff
@@ -45,100 +46,117 @@ class _DisplayCardState extends State<DisplayCard> {
...
+            child: Row(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: <Widget>[
+                  Column(children: <Widget>[
+                    // Title
+                    Text(
+                      (widget.title.length > 35)
+                          ? widget.title.substring(0, 35)
+                          : widget.title,
```
Comment (Cooljawty) [inline app/lib/uiwidgets/cards.dart:115]: I think It would be better to restrict the length of a title instead of just displaying a truncated version.

**Review by Cooljawty — CHANGES_REQUESTED** [review]
Once again really great job on this one. The only change I suggested is setting the maximum length of titles instead of only displaying it partialy.

File: app/lib/uiwidgets/cards.dart (hunk 2 above)
Comment (Benek6h) [reply app/lib/uiwidgets/cards.dart]: I feel that users should be able to make any title they want. The idea of being more restrictive than a real journal betrays what we had in mind.

Comment (Cooljawty) [reply app/lib/uiwidgets/cards.dart]: Ok yeah I can agree with that. That was my only complaint

**Review by Cooljawty — APPROVED** [review]

**Pull request merged**